added enable and disable wifi commands

mangiari opened the pull request:
I added commands for closing the wifi, so that the player can be used offline by a wave of a rfid card. Code is untested by now, but should not do anything, until configured to a card. No clue if the phoniebox has a problem when having no network.

MiczFlor commented:
Hi @mangiari 
thanks for the addition. I am thinking about the use case. Why would you need this feature? All the best and keep the code coming :)
micz

mangiari commented:
Hi micz,
I only need the Wifi to put new music on the SD, or to do other administrative stuff. When the baby boy is playing with the phoniebox I would rather not want it sending wifi. 
Most people that do real research (not voodoo) about effects of electromagnetic radiation, tend to not have wifi at home.
I have WiFi, but I try only to have it sendind when it is actually in use. The main network and also the clients.
I also wanted to disconnect the RFID reader as long as not needed, but my zero w crashes and reboots when I connect the RFID reader back to the board. Pulling the cable is no problem. But switching back on takes way to long until the reading is ready (because of the reboot)
Thanks for adding.
I saw you were active yesterday, there was a time slot when the register card was broken after I did a pull. Better work with development/feature branches before pushing, that's how we do in the office... ;)
Grüße aus München,
Frank
Am 22.08.2018 23:29 schrieb Micz Flor <notifications@github.com>:Hi @mangiari
thanks for the addition. I am thinking about the use case. Why would you need this feature? All the best and keep the code coming :)
micz

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

zxa commented:
> Most people that do real research (not voodoo) about effects of electromagnetic radiation, tend to not have wifi at home.

Do you have any evidence to back up that claim?

MiczFlor commented on scripts/playout_controls.sh:
```diff
@@ -365,6 +367,14 @@ case $COMMAND in
+	enablewifi)
+		rfkill unblock wifi
+		;;
+	disablewifi
```
Hi @mangiari 
I get it now. And it makes sense. And I believe that electro smog might eventually turn out to be what radio activity was only decades ago: an invisible and therefore underestimated hazzard.
However, please check the code an then create a pull request again. There is an `)` missing in this line.
Thanks, micz

mangiari commented:
Sorry about my syntax issue, I even found another one. I'm typically writing java code in an IDE and it happens very seldom that it is not ok after the first try. You know in place compilation...

I corrected the stuff, added some more docu also to the manual and now I also tested it. I can't promise that there is no radiation from the wlan any more, but at least it is not working any more and when I wave the card with the enablewifi command it is up and running again. I think it's also dependent on the hardware whether it is really shut down or just not accessible for the software.

Luegengladiator commented:
I like the feature. Special for mobile phonies (usage in the car) we can save some power by not scanning for WIFI Networks.

MiczFlor commented:
Hi @mangiari 
back on my desk. Thanks for the addition. Please don't get confused that I changed the base to `develop`. By now there are too many additions to all merge or push them straight to `master` - as you pointed out yourself :)
P.S.: did you add this to the documentation? If not: could you make a pull request for that, too?
Thanks

mangiari commented:
Yes added to the manual and other missing commands also.
Am 30.08.2018 09:42 schrieb Micz Flor <notifications@github.com>:Hi @mangiari
back on my desk. Thanks for the addition. Please don't get confused that I changed the base to develop. By now there are too many additions to all merge or push them straight to master - as you pointed out yourself :)
P.S.: did you add this to the documentation? If not: could you make a pull request for that, too?
Thanks

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

gizmo21 commented:
Hi there, like the feature and disable (rfkill block) works a far as I can see on my USB-Wifi-Dongle ("EDIMAX EW-7811UN Wireless USB Adapter") on PI 2 B as the lights on dongle turn off and rfkill list shows:
0: phy0: Wireless LAN
        Soft blocked: yes
        Hard blocked: no

But unblocking doesn't work, well hardware seems to be activated again as dongle-LED flickers again and rfkill list shows:
```
0: phy0: Wireless LAN
        Soft blocked: no
        Hard blocked: no
```
but I get no connection even with following lines added to the script: 
```
 sudo ifconfig wlan0 up
 sudo ip link set wlan0 up
sudo networking restart

```
`ifconfig wlan0`  still show no link/ip adress until I restart device.

Any advice how to trigger networking to restart my unblocked usb dongle on Raspbian GNU/Linux 9 (stretch)

mangiari commented:
Hm, said. On my infernal wifi of the zero w and also the Raspberry pi 3, it works smooth on and off and on again. Did not check any further and also do not have an external WiFi to test any changes/improvements, sorry 
Am 23.09.2018 2:54 nachm. schrieb gizmo21 <notifications@github.com>:Hi there, like the feature and disable (rfkill block) works a far as I can see on my USB-Wifi-Dongle ("EDIMAX EW-7811UN Wireless USB Adapter") on PI 2 B as the lights on dongle turn off and rfkill list shows:
0: phy0: Wireless LAN
Soft blocked: yes
Hard blocked: no
But unblocking doesn't work, well hardware seems to be activated again as dongle-LED flickers again and rfkill list shows:
0: phy0: Wireless LAN
        Soft blocked: no
        Hard blocked: no

but I get no connection even with following lines added to the script:
 sudo ifconfig wlan0 up
 sudo ip link set wlan0 up
sudo networking restart


ifconfig wlan0  still show no link/ip adress until I restart device.
Any advice how to trigger networking to restart my unblocked usb dongle on Raspbian GNU/Linux 9 (stretch)

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.

gizmo21 commented:
No problem, perhaps someone else has an idea how to reconnect an USB wif-dongle after rfkill block.

gusmax777 commented:
I know that this is an old post, but may I add a request to this feature. The basic idea to turn off the wifi of the box is great to avoid radiation. What I do not like is, that I would need to switch the wifi of manually. Why do not switch it off automatically after 30 minutes after a reboot or last activation by the card. To avoid to have to reactivate the Wifi every 30 minutes a switch in the GUI could be helpfull to leave the wifi on until the next reboot. What do you think about this?
Cheers, Thomas

mangiari commented:
I think it should be a config setting whether Wifi should be enabled or disabled at startup.
GUI button for toggle is difficult, but GUI button for disabling sounds nice.
Am 12.12.2018 22:31 schrieb gusmax777 <notifications@github.com>:I know that this is an old post, but may I add a request to this feature. The basic idea to turn off the wifi of the box is great to avoid radiation. What I do not like is, that I would need to switch the wifi of manually. Why do not switch it off automatically after 30 minutes after a reboot or last activation by the card. To avoid to have to reactivate the Wifi every 30 minutes a switch in the GUI could be helpfull to leave the wifi on until the next reboot. What do you think about this?
Cheers, Thomas

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or mute the thread.
